### ISSUE TITLE
The order of aliases in wait is no longer respected with conditional aliases

### ISSUE THREAD

**Issue by freon27** [cypress-io/cypress#28949]
### Current behavior

We use GraphQL in the way recommended in the Cypress docs. IE with a single interceptor and conditional setting of the aliases. This worked correctly up until 13.3.1. 

Since then the order of responses yielded after waiting on an array of aliases no longer respects the defined order.

It appears that the bug was introduced by this PR. I've written a test in my fork of Cypress on the commit for 13.3.1 and then rolled the change back manually and the test passes.

https://github.com/cypress-io/cypress/pull/27806

### Desired behavior

The order of responses should match the order the aliases was passed to await.

I would be interested in helping with a fix. I had made a start on this but I'm struggling to understand how to retrieve the alias for each request.

### Test code to reproduce

This fork uses fetch instead of GraphQL but the principle is the same. If you look at the commit history, I have tested it with 13.3.0 and 13.3.1 and the test passes in 13.3.0.

https://github.com/freon27/cypress-test-tiny

### Cypress Version

everything above 13.3.1 is broken

### Node version

18.18

### Operating System

MacOS 14.1.1

**Comment by jennifer-shehane** [comment]
@freon27 Could you give instructions on how to run this repo you provided?

**Comment by freon27** [comment]
> @freon27 Could you give instructions on how to run this repo you provided?

Sorry... I mean to do this and forgot.

I've pushed one more commit, so if you could pull that and then do `npm run start` to start the server and `npm run cypress:open` to see the test.

**Comment by jennifer-shehane** [comment]
This example does not work because all of the requests were intercepted via one intercept definition, so they all have the same `routeId`, which is what's being used to determine uniqueness here: https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cy/commands/waiting.ts#L252

![Screenshot 2024-02-19 at 12 35 39 PM](https://github.com/cypress-io/cypress/assets/1271364/a14167f1-f617-4b87-a0c0-128ac98f3b65)
![Screenshot 2024-02-19 at 12 35 20 PM](https://github.com/cypress-io/cypress/assets/1271364/b0ca2779-41db-4ae6-bb96-8080e755d155)


For now, you could break our your intercept, otherwise we'd accept a PR to handle this in a better way in this case.

**Comment by jennifer-shehane** [comment]
An example in our repo that would fail btw:

```js
it('all responses returned in correct order - using same intercept definition, () => {
        cy.intercept('*', (req) => {
          if (req.url.includes('alpha')) {
            req.alias = 'getAlpha'
            req.continue((res) => {
              res.send({ value: 'alpha' })
            })
          } else if (req.url.includes('beta')) {
            req.alias = 'getBeta'
            req.continue((res) => {
              res.send({ value: 'beta' })
            })
          } else if (req.url.includes('gamma')) {
            req.alias = 'getGamma'
            req.continue((res) => {
              res.send({ value: 'gamma' })
            })
          } else if (req.url.includes('delta')) {
            req.alias = 'getDelta'
            req.continue((res) => {
              res.send({ value: 'delta' })
            })
          } else if (req.url.includes('epsilon')) {
            req.alias = 'getEpsilon'
            req.continue((res) => {
              res.send({ value: 'epsilon' })
            })
          }
        })

        cy.window().then((win) => {
          xhrGet(win, '/epsilon')
          xhrGet(win, '/beta')
          xhrGet(win, '/gamma')
          xhrGet(win, '/delta')
          xhrGet(win, '/alpha')

          return null
        })

        cy.wait(['@getAlpha', '@getBeta', '@getGamma', '@getDelta', '@getEpsilon']).then((responses) => {
          expect(responses[0]?.response?.body.value).to.eq('alpha')
          expect(responses[1]?.response?.body.value).to.eq('beta')
          expect(responses[2]?.response?.body.value).to.eq('gamma')
          expect(responses[3]?.response?.body.value).to.eq('delta')
          expect(responses[4]?.response?.body.value).to.eq('epsilon')
        })
      })
```

**Comment by freon27** [comment]
@jennifer-shehane Thanks for the responses :) 

The problem is that it **did** work in the past so it seems to me to be a regression caused in 13.3.1. Our entire test suite is now written based on this working so we are stuck on 13.3.0.

I don't actually see how I can use multiple interceptors with GraphQL. We're using this solution: https://docs.cypress.io/guides/end-to-end-testing/working-with-graphql. 

That relies on a single interceptor with conditional code inside, so we have always understood this to be a "safe" solution. 

I wrote exactly the test you've listed when I was looking for a solution myself. I got stuck trying to work out how to get back to the aliases from the route IDs. I had tried using `orderedAliases` from the state. It worked in my test but then in other tests the list was empty.

I think the ideal solution is to group by alias and then only sort within the alias groups. If someone could help me understand how to to get back to the aliases then I could work on a PR.
